app/models/katello/concerns/content_aspect_host_extensions.rb

```diff
@@ -0,0 +1,22 @@
+module Katello
+  module Concerns
+    module ContentAspectHostExtensions
+      extend ActiveSupport::Concern
+
+      included do
+        has_one :content_aspect, :class_name => '::Katello::Host::ContentAspect', :foreign_key => :host_id, :inverse_of => :host, :dependent =>  :destroy
+        has_one :content_view, :through => :content_aspect
+        has_one :lifecycle_environment, :through => :content_aspect
```

Comment:
Actually this is required for scoped search, at least it simplifies it greatly.  I could probably avoid it using https://github.com/wvanbergen/scoped_search/wiki/Usage#external-search-methods

but this is simpler and easier to understand.  I have updated other usages to go through the aspect manually though.